lib/sneakers/worker.rb

```diff
@@ -61,11 +61,11 @@ def do_work(delivery_info, metadata, msg, handler)
           end
         rescue Timeout::Error
           res = :timeout
-          worker_error('timeout')
+          worker_error('timeout',msg)
         rescue => ex
           res = :error
           error = ex
-          worker_error('unexpected error', ex)
+          worker_error('unexpected error', msg, ex)
```

Comment:
The `raven-ruby` gem is what Sentry uses to hook into our apps, and it includes a [shim](https://github.com/jondot/sneakers/blob/master/lib/sneakers/worker.rb#L66-L68) for Sidekiq:
```ruby
raven-ruby/lib/raven/integrations/sidekiq.rb
Sidekiq.configure_server do |config|
  config.error_handlers << Proc.new {|ex,context| Raven.capture_exception(ex, :extra => {:sidekiq => context}) }
end
```
And then Sidekiq has a [method](https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/exception_handler.rb#L20) that's invoked in their processing rescues that enumerates the `config.error_handlers` collection and passes the context hash to the interface method:
```ruby
# sidekiq/lib/sidekiq/exception_handler.rb
def handle_exception(ex, ctxHash={})
  Sidekiq.error_handlers.each do |handler|
    begin
      handler.call(ex, ctxHash)
    rescue => ex
      Sidekiq.logger.error "!!! ERROR HANDLER THREW AN ERROR !!!"
      Sidekiq.logger.error ex
      Sidekiq.logger.error ex.backtrace.join("\n") unless ex.backtrace.nil?
    end
  end
end
```
Sneakers should use a similar pattern. [`Sneakers::Worker#worker_error`](https://github.com/jondot/sneakers/blob/master/lib/sneakers/worker.rb#L112-L120) should should enumerate the low-level error handlers (maybe "reporters" instead to avoid ambiguity with the already existing sneakers handlers) collection instead of logging things:
```ruby
# sneakers/lib/sneakers/worker.rb
def worker_error(msg, exception)
  Sneakers.error_reporters.each do |reporter|
    reporter.call(exception, { message: msg }) # add whatever else could be help for the worker context
  end
end
```
We could include a default reporter that copied sidekiq's default error [handler](https://github.com/mperham/sidekiq/blob/7a727c61cfd3f174d32e3a231b14833b61b897fe/lib/sidekiq/exception_handler.rb#L6-L15) and would log the message like you are wanting:
```
# sneakers/lib/sneakers/exception_reporter.rb
module Sneakers
  module ExceptionReporter
    class Logger
      def call(exception, context = {})
        Sneakers.logger.warn(context) if !context.empty?
        Sneakers.logger.warn "#{exception.class.name}: #{ex.message}"
        Sneakers.logger.warn exception.backtrace.join("\n") unless exception.backtrace.nil?
      end
    end
  end
end
```